python/ray/ml/train/data_parallel_ingest.py

```diff
@@ -0,0 +1,70 @@
+from typing import Dict, List, Union, TYPE_CHECKING
+
+from ray.actor import ActorHandle
+from ray.ml.config import DatasetConfig
+from ray.ml.preprocessor import Preprocessor
+
+if TYPE_CHECKING:
+    from ray.data import Dataset, DatasetPipeline
+
+
+class _DataParallelIngestSpec:
+    def __init__(
+        self, datasets: Dict[str, "Dataset"], dataset_config: Dict[str, DatasetConfig]
+    ):
+        self.datasets = datasets
+        self.dataset_config = dataset_config
+
+    def preprocess_datasets(self, prep: Preprocessor, datasets: Dict[str, "Dataset"]):
+        if not prep:
+            return datasets
+        ds_to_fit = None
+        for k, v in self.dataset_config.items():
+            if v.fit:
+                ds_to_fit = v
+        if ds_to_fit:
+            prep.fit(ds_to_fit)
```

Comment:
`ds_to_fit` is a `DatasetConfig` right? Should this be `prep.fit(self.datasets[k])`?

Reply:
Fixed.